tests/test_sagemaker/test_sagemaker_pipeline.py

```diff
@@ -0,0 +1,176 @@
+from moto import mock_sagemaker
+from time import sleep
+from datetime import datetime
+import boto3
+import botocore
+import pytest
+
+from moto.core import DEFAULT_ACCOUNT_ID as ACCOUNT_ID
+
+FAKE_ROLE_ARN = f"arn:aws:iam::{ACCOUNT_ID}:role/FakeRole"
+TEST_REGION_NAME = "us-east-1"
+
+
+def arn_formatter(_type, _id, account_id, region_name):
+    return f"arn:aws:sagemaker:{region_name}:{account_id}:{_type}/{_id}"
+
+
+@pytest.fixture(name="sagemaker_client")
+def fixture_sagemaker_client():
+    with mock_sagemaker():
+        yield boto3.client("sagemaker", region_name=TEST_REGION_NAME)
+
+
+def create_sagemaker_pipelines(sagemaker_client, pipeline_names, wait_seconds=0.0):
+    responses = []
+    for pipeline_name in pipeline_names:
+        responses += sagemaker_client.create_pipeline(
+            PipelineName=pipeline_name,
+            RoleArn=FAKE_ROLE_ARN,
+        )
+        sleep(wait_seconds)
+    return responses
+
+
+def test_create_pipeline(sagemaker_client):
+    fake_pipeline_name = "MyPipelineName"
+    response = sagemaker_client.create_pipeline(
+        PipelineName=fake_pipeline_name,
+        RoleArn=FAKE_ROLE_ARN,
+    )
+    assert isinstance(response, dict)
+    assert response["PipelineArn"].should.equal(
+        arn_formatter("pipeline", fake_pipeline_name, ACCOUNT_ID, TEST_REGION_NAME)
+    )
+
+
+def test_list_pipelines_none(sagemaker_client):
+    response = sagemaker_client.list_pipelines()
+    assert isinstance(response, dict)
+    assert response["PipelineSummaries"].should.be.empty
```

Comment:
```suggestion
    response["PipelineSummaries"].should.be.empty
```
If you use the `should`, you don't need to assert - asserting whatever it `should` be is automatic.

There are quite a few assertions like that - I won't point them all out, but ideally we remove all of them.

Reply:
flake8 only gives me a pass when having an `assert` before `.should.be.empty` statements. For other `should.have` (e.g. `should.have.length_of`), no assert seems to be required.

See here the workflow run on my forked repo: https://github.com/stiebels/moto/actions/runs/3709067440/jobs/6287306243

Therefore, I've left the `assert` in for `should.be.empty` and removed it where it doesn't seem to be needed.